io/jvm/src/main/scala/scalaz/effect/RTS.scala

```diff
@@ -557,16 +558,16 @@ private object RTS {
                     } finally enterAsyncEnd()
 
                   case IO.Tags.Attempt =>
-                    val io = curIo.asInstanceOf[IO.Attempt[E, Any, Any]]
+                    val io = curIo.asInstanceOf[IO.Attempt[E, Any, Any, Any]]
 
                     curIo = io.value
 
-                    stack.push(Catcher)
+                    stack.push(new Catcher(io.err, io.succ))
```

Comment:
We can eliminate this allocation. Instead of using `Catcher` here, let's just use `Attempt`, by making `Attempt` extend `A => IO[E2, B]`. Then you push the catcher onto the stack by pushing the `Attempt` (which becomes the new `Catcher`). That's 1 fewer allocations and will bring it inline with the old model.

Reply:
Sweet. Will do it.